src/CMakeLists.txt

```diff
@@ -400,6 +400,33 @@ if(NOT CMAKE_CROSSCOMPILING)
         ${CARGO} test --all-features --manifest-path ${CMAKE_CURRENT_SOURCE_DIR}/rust/Cargo.toml --target-dir ${RUST_BINARY_DIR} ${RUST_CARGO_FLAG}
     )
   add_dependencies(rust-test rust-bindgen)
+
+  add_custom_target(rust-clippy
+    COMMAND
+      # Force clippy to fully re-run. It is bad at figuring out when to run again and when to use caches.
+      ${CARGO} clean --manifest-path ${CMAKE_CURRENT_SOURCE_DIR}/rust/Cargo.toml --target-dir ${RUST_BINARY_DIR}
+    COMMAND
+      ${CMAKE_COMMAND} -E env
+        CMAKE_CURRENT_BINARY_DIR=${CMAKE_CURRENT_BINARY_DIR}
+        # enable nightly features in the stable release - needed to activate alloc_error_handler.
```

Comment:
Yeah, not sure if clippy compiles or parses or whatever, but it will also complain about the illegal stuff same as `cargo build`, unless this is here.